nft_ingester/src/price_fetcher.rs

```diff
@@ -8,11 +8,22 @@ use rocks_db::Storage;
 use std::sync::Arc;
 use std::time::Duration;
 use tokio::sync::broadcast::Receiver;
-use tracing::log::error;
+use tokio::task::JoinError;
+use tracing::log::{error, info};
 
 pub const SOLANA_CURRENCY: &str = "solana";
 const USD_CURRENCY: &str = "usd";
 
+pub async fn start_price_monitoring<T: PriceFetcher>(
```

Comment:
The same concern as described above. We call this function once, albeit we have a wrapper for it. Could you explain the idea behind it, please?